File: Java Assignment/BankAccountInt/src/Account.java
import Exceptions.InsufficientFundsException;
import Exceptions.InvalidAmountException;
import Exceptions.InvalidPinException;

public class Account {

    private String name;
    private int balance;
    private String pin;
    private String accNo;

    public Account(String name, String accNo, int balance, String pin){

        this.name = name;
        this.accNo = accNo;
        this.balance = balance;
        this.pin = pin;

    }


    public void deposit(int amount) {
        if(amount > 0) {
            balance += amount;
        }
    }

    public void withdraw(int amount, String pin) {
        if(!this.pin.equals(pin)){
            throw new InvalidPinException("Invalid pin");
        }
        if(amount < 0){
            throw new InvalidAmountException(("Enter a valid amount"));
        }
        if(balance < amount){
            throw new InsufficientFundsException("Insufficient amount");
        }
        balance -= amount;
    }

    public int checkBalance(String accNO) {
        return balance;
    }
}

File: Python assingment/BankApp/account.py
class Account:
    def __init__(self,name, acc_no, balance, pin):
        self.name = name
        self.acc_no = acc_no
        self.balance = balance
        self.pin = pin

    def deposit(self, amount):

        if amount > 0:
            self.balance += amount
            return self.balance

        raise ValueError ("Invalid amount")


    def withdraw(self, amount, pin):

        if pin != self.pin:
            raise ValueError ("Invalid pin")

        if amount > self.balance:
            return ValueError ("Insufficient funds")

        self.balance -= amount

        return self.balance


    def check_balance(self, pin):

        if pin != self.pin:
            raise ValueError ("Invalid pin")

        return self.balance



























